src/IO/readFloatText.h

```diff
@@ -303,10 +313,16 @@ static inline void readUIntTextUpToNSignificantDigits(T & x, ReadBuffer & buf)
             else
                 return;
         }
+    }
 
-        while (!buf.eof() && isNumericASCII(*buf.position()))
-            ++buf.position();
+    while (!buf.eof() && (buf.position() + 8 <= buf.buffer().end()) &&
```

Comment:
We can also do it for up to 16 digits with SSE2/SSE4.2:
- check that every character in SSE register is in [0..9];
- find position of the first non-matching character.

Reply:
Nice. Will do.

Reply:
@alexey-milovidov  After implementation using isNext16Digits does not make performance improvements.